js/interface.js

```diff
@@ -114,21 +130,49 @@ var dsQueryProvider = Fliplet.Widget.open('com.fliplet.data-source-query', {
 });
 
 dsQueryProvider.then(function onForwardDsQueryProvider(result) {
-  var validation = {
-    sms: {
-      toColumn: result.data.columns.smsTo,
-      matchColumn: result.data.columns.smsMatch,
-      template: $('#sms-template').val(),
-      expire: $('#expire-timeout').val(),
-    },
-    email: {
-      toColumn: result.data.columns.emailMatch,
-      matchColumn: result.data.columns.emailMatch,
-      template: emailProviderResult || defaultEmailSettings,
-      expire: $('#expire-timeout').val(),
-    }
+  // If there is no data source just save it althoug not valid config
+  if (!dataSource) {
+    Fliplet.Widget.save();
   }
 
+  var validation = dataSource.definition.validation || {};
+
+  // Let's update only the selected verification type on the datasurce settings
+  switch(data.type) {
+    case 'sms':
+      validation.sms = {
+        toColumn: result.data.columns.smsTo,
+        matchColumn: result.data.columns.smsMatch,
+        template: $('#sms-template').val(),
+        expire: $('#expire-timeout').val(),
+      };
+      break;
+
+    case 'email':
+      validation.email = {
+        toColumn: result.data.columns.emailMatch,
+        matchColumn: result.data.columns.emailMatch,
+        template: emailProviderResult || defaultEmailSettings,
+        expire: $('#expire-timeout').val(),
+      }
+      break;
+
+    case 'domain':
```

Comment:
Somehow I need to store the settings under this setting on the data source definition...
We need to make it like that so when you are requesting the code to the server we know if you are using just email or domain. They don't work in the same way on the server.